etrago/tools/utilities.py

```diff
@@ -145,11 +145,17 @@ def geolocation_buses(self):
 
     Parameters
     ----------
-    etrago : :class:`etrago.Etrago`
+    self : :class:`etrago.Etrago`
        Transmission grid object
+       or
+       :class:'pypsa.components.Network'
 
     """
-    network = self.network
+
+    if not isinstance(self, pypsa.components.Network):
+        network = self.network
+    else:
+        network = self
```

Comment:
Allowing different data types seems a bit dangerous, if it is possible to avoid things like this, I would do it. 
It also is a bit strange because this function is part of an object, but also works without it. 

Was there any reason why you did not change that function in a similar way as you changed the buses_by_country function? I guess this one could also operate on the network only.